src/main/resources/com/google/api/codegen/js/sample.snip

```diff
@@ -0,0 +1,172 @@
+@snippet generate(class)
+  <!--
+  BEFORE RUNNING:
+  ---------------
+  1. If not already done, enable the {@class.apiTitle}
+     and check the quota for your project at
+     https://console.developers.google.com/apis/api/{@class.apiName}
+  2. Get access keys for your application. See
+     https://developers.google.com/api-client-library/javascript/start/start-js#get-access-keys-for-your-application
+  -->
+  <script src="https://apis.google.com/js/api.js"></script>
+  <script>
+  function {@class.makeApiCallFuncName}() {
+    @if class.needParams
+      var {@class.paramsVarName} = {
+        @join field : class.fields if class.fields on BREAK.add(BREAK)
+          @if field.description
+            @join line : util.getDocLines(field.description, 100)
+              // {@line}
+            @end
+
+
+          @end
+          {@field.name}: {@field.defaultValue},  // TODO: Update placeholder value.
+          @if field.example
+            // {@field.example}
+          @end
+        @end 
+      };
+
+    @end
+    @if class.hasRequestBody
+      var {@class.requestBodyVarName} = {
+        @switch class.methodVerb
+        @case "PATCH"
+          // TODO: Add desired properties to the request body. Only these properties
+          // will be changed.
+        @case "PUT"
+          // TODO: Add desired properties to the request body. All existing properties
+          // will be replaced.
+        @default
+          // TODO: Add desired properties to the request body.
+        @end
+      };
+
+    @end
+    @if class.hasMediaUpload
+      // TODO: To do media uploads, it's more appropriate to use CORS. See
+      // https://developers.google.com/api-client-library/javascript/features/cors
+
+    @end
+    @if class.hasMediaDownload
+      // TODO: To do media downloads, it's more appropriate to use CORS. See
+      // https://developers.google.com/api-client-library/javascript/features/cors
+
+    @end
+    @if class.isPageStreaming
+      {@class.pageStreaming.executeRequestFuncName}({@argList(class, @FALSE)});
+    @else
+      var {@class.requestVarName} = gapi.client.{@class.apiName}.{@callChain(class.methodNameComponents)}({@argList(class, @FALSE)});
+      @if class.hasResponse
+        {@class.requestVarName}.then(function({@class.responseVarName}) {
+          // TODO: Change code below to process the `{@class.responseVarName}` object:
+          console.log({@class.responseVarName}.result);
```

Comment:
Good point, done. I copied this mostly from the Node sample... so I'll make a CL to do it there as well.